platforms/HomeAssistant.js

```diff
@@ -0,0 +1,301 @@
+// Home Assistant
+//
+// Current Support: lights
+//
+// This is a shim to publish lights maintained by Home Assistant.
+// Home Assistant is an open-source home automation platform.
+// URL:     http://home-assistant.io
+// GitHub:  https://github.com/balloob/home-assistant
+//
+// Remember to add platform to config.json. Example:
+// "platforms": [
+//    {
+//        "platform": "HomeAssistant",
+//        "name": "HomeAssistant",
+//        "host": "http://192.168.1.50:8123",
+//        "password": "xxx"
+//    }
+// ]
+//
+// When you attempt to add a device, it will ask for a "PIN code".
+// The default code for all HomeBridge accessories is 031-45-154.
+
+var types = require("HAP-NodeJS/accessories/types.js");
+var url = require('url')
+var request = require("request");
+
+function HomeAssistantPlatform(log, config){
+
+  // auth info
+  this.host = config["host"];
+  this.password = config["password"];
+
+  this.log = log;
+}
+
+HomeAssistantPlatform.prototype = {
+  _request: function(method, path, options, callback) {
+    var self = this
+    var requestURL = this.host + '/api' + path
+    options = options || {}
+    options.query = options.query || {}
+
+    var reqOpts = {
+      url: url.parse(requestURL),
+      method: method || 'GET',
+      qs: options.query,
+      body: JSON.stringify(options.body),
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+        'x-ha-access': this.password
+      }
+    }
+
+    request(reqOpts, function onResponse(error, response, body) {
+      if (error) {
+        callback(error, response)
+        return
+      }
+
+      if (response.statusCode === 401) {
+        callback(new Error('You are not authenticated'), response)
+        return
+      }
+
+      json = JSON.parse(body)
+      callback(error, response, json)
+    })
+
+  },
+  accessories: function(callback) {
+    this.log("Fetching HomeAssistant devices.");
+
+    var that = this;
+    var foundAccessories = [];
+    var lightsRE = /^light\./i
+
+
+    this._request('GET', '/states', {}, function(error, response, data){
+
+      for (var i = 0; i < data.length; i++) {
+        entity = data[i]
+
+        if (entity.entity_id.match(lightsRE)) {
+          accessory = new HomeAssistantAccessory(that.log, entity, that)
+          foundAccessories.push(accessory)
+        }
+      }
+
+      callback(foundAccessories)
+    })
+
+  }
+}
+
+function HomeAssistantAccessory(log, data, client) {
+  // device info
+  this.data = data
+  this.entity_id = data.entity_id
+  this.name = data.attributes.friendly_name
```

Comment:
Not every entity will have this attribute. In [the HA frontend](https://github.com/balloob/home-assistant-js/blob/master/src/modules/entity/model.js#L40) I fall back to the object_id (part right of the `.` in entity_id) with `_` replaced by `<space>`